Assets/Scripts/DoorMechanism.cs

```diff
@@ -7,15 +7,46 @@ public class DoorMechanism : MonoBehaviour
     Animator animator;
     bool doorOpen;
 
+    // Added array of GameObjects to keep track of number of enemies.
+    // enemyCount just to simplify the condition in OnTriggerEnter
+    public GameObject player;
+    public GameObject[] enemies;
+    public int enemyCount = 0;
+
     void Start()
     {
         doorOpen = false;
         animator = GetComponent<Animator>();
+
+        // initializes enemies array and sets enemyCount 
+        // for number of enemies in the scene
```

Comment:
Just want to note that since this is counting all enemies in the scene, we will be unable to have multiple rooms in 1 scene.

Reply:
Maybe we can attach a script to each Room prefab instead, where the Room keeps count of how many enemies there are within it? Each time an enemy's health reaches zero and it becomes eligible for possession, we should then decrement the count. Add an event listener in the Room script so that if the count hits zero, the Room triggers all its doors to unlock.

The problem I foresee is with multiple Rooms adjacent to one another. With the current design, we'll have two doors next to each other. One will rise because the room was cleared; the other will not because it belongs to another Room. Perhaps we should separate the doors/walls from the Room itself? Let me know what you think, Jazy.